pkg/activator/test/roundtripper.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package test
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+
+	"github.com/knative/serving/pkg/network"
+	"github.com/knative/serving/pkg/queue"
+)
+
+// FakeResponse is a response given by the FakeRoundTripper
+type FakeResponse struct {
+	Err  error
+	Code int
+	Body string
+}
+
+// FakeRoundTripper is a roundtripper emulator useful in testing
+type FakeRoundTripper struct {
+	T *testing.T
+
+	// Return an error if host header does not match this
+	ExpectHost string
+
+	// lockerCh blocks responses being sent until a struct is written to the channel
+	LockerCh chan struct{}
+
+	// Responses to probe requests are popeed from this list until it is size 1 then
+	// that response is returned indefinitely
+	ProbeResponses []FakeResponse
+
+	// Response to non-probe requests
+	RequestResponse *FakeResponse
+
+	mux sync.Mutex
```

Comment:
What's this mux guarding? Seems like it's only `ProbeResponses`. Maybe move it right above/below the thing it's guarding and name it accordingly, like `probeResponsesMux`. It's a lot easier to read that vs. having to search for usages through the file.